src/Giraffe.HotReload/Library.fs

```diff
@@ -206,28 +212,46 @@ module rec LiveUpdate =
   /// A middleware that delegates to GiraffeMiddlware, but updates the middleware when the HttpHandler is updated
   type HotReloadGiraffeMiddleware(next: RequestDelegate,
                                   handler: HttpHandler,
+                                  hostEnv : IHostingEnvironment,
                                   sockets: ResizeArray<System.Net.WebSockets.WebSocket>,
                                   settings : Settings,
                                   loggerFactory: ILoggerFactory) as self =
         let logger = loggerFactory.CreateLogger<HotReloadGiraffeMiddleware>()
         let merge handler = choose [LiveUpdate.updater settings self; handler ]
         let refreshCommand = "refresh" |> System.Text.Encoding.UTF8.GetBytes |> ArraySegment
-        let mutable innerMiddleware = Middleware.GiraffeMiddleware(next, merge handler, loggerFactory)
-
-        member __.Invoke (ctx: HttpContext) = innerMiddleware.Invoke ctx
-
-        /// Replace the current giraffe pipeline with a new one based on the provided `HttpHandler`
-        member __.Update(newHandler) =
-          innerMiddleware <- Middleware.GiraffeMiddleware(next, merge newHandler, loggerFactory)
+        let sendRefreshCommand (socket : WebSocket) = socket.SendAsync(refreshCommand, Net.WebSockets.WebSocketMessageType.Text, true, Threading.CancellationToken.None)
+        let notifyAllSockets sockets =
           sockets
           |> Seq.mapi (fun i socket -> task {
               logger.LogInformation("Notifying websocket {id}", i)
-              do! socket.SendAsync(refreshCommand, Net.WebSockets.WebSocketMessageType.Text, true, Threading.CancellationToken.None)
+              do! sendRefreshCommand socket
             }
           )
           |> System.Threading.Tasks.Task.WhenAll
           |> ignore
 
+        let handleFileChange () =
+          logger.LogDebug("Static content file changed. Sending refresh commands")
```

Comment:
Unfortunately with the `fileProvider.Watch` it doesn't give feedback as to which file changed. There are many issues like https://github.com/aspnet/Extensions/issues/846 opened around the many repos.